docs/pages/management/dynamic-configuration.mdx

```diff
@@ -0,0 +1,272 @@
+  end 
+  subgraph proxy[Proxy Service]
+    conf2["Static config"]
+  end 
+end
+
+auth--"Reads, writes"-->backend
+subgraph clients["API clients"]
+  terraform["Terraform Provider"]
+  operator["Kubernetes Operator"]
+  tctl
+end
+terraform & operator & tctl-- "API calls" -->proxy
+proxy-.->auth
+```
+
+Teleport borrows the concept of a dynamic configuration resource from the REST
+programming pattern. A cluster is composed of different objects (i.e.,
+resources) and there are three common operations that can be performed on them:
+`get` , `create` , and `remove` .
+
+Every resource in Teleport has three required fields:
+
+- `kind`: The type of resource
+- `name`: A required field in the `metadata` to uniquely identify the resource
+- `version`: The version of the resource format
+
+All other fields are specific to a resource.
+
+While Teleport Enterprise Cloud and Teleport Team do not expose the static
+configuration file to operators, they do use a static configuration file for
+certain settings. Read how Teleport [reconciles static and dynamic
+resources](#reconciling-the-configuration-file-with-dynamic-resources) to
+understand how to see the values of static configuration settings that also
+appear in dynamic configuration resources.
+
+When examining a dynamic configuration resource, note that some of the fields
+you will see are used only internally and are not meant to be changed. Others
+are reserved for future use.
+
+## Managing dynamic configuration resources
+
+Teleport provides three methods for applying dynamic configuration resources:
+the `tctl` client tool, Teleport Terraform provider, and Kubernetes Operator.
+
+All three methods connect to the Teleport Auth Service's gRPC endpoint in order
+to manipulate cluster configuration resources stored on the Auth Service
+backend. The design of Teleport's configuration interface makes it well suited
+for infrastructure-as-code and GitOps approaches.
+
+### YAML documents with `tctl`
+
+You can define configuration resources as YAML documents and apply them using
+the `tctl` client tool. Here is an example of a `role` resource that allows
+access to servers with the label `env:test`:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: developer
+spec:
+  allow:
+    logins: ['ubuntu', 'debian', '{{internal.logins}}']
+    node_labels:
+      'env': 'test'
+```
+
+Since `tctl` works from the local filesystem, you can write commands that apply
+all configuration documents in a directory tree. See the [CLI
+reference](../reference/cli.mdx#tctl) for more information on `tctl`.
+
+### Teleport Terraform provider
+
+Teleport's Terraform provider lets you manage your Teleport configuration
+resources within the same infrastructure-as-code source as the rest of your
+infrastructure. There is a Terraform resource for each Teleport configuration
+resource. For example:
+
+```text
+resource "teleport_role" "developer" {
+  metadata = {
+    name        = "developer"
```

Comment:
To match the last two keys:

```suggestion
    name      = "developer"
```

Or to match the middle key:

```suggestion
    name     = "developer"
```

Or to avoid trying to match:

```suggestion
    name = "developer"
```

Reply:
I've used the formatting that `terraform fmt` uses, so I think it makes sense to keep it this way. This is what a user would see if they were to use `terraform fmt` on their own config